processor/commonproc/preload.go

```diff
@@ -41,16 +44,19 @@ func (*extractPreloadHeaders) Process(resp *exchange.Response) error {
 	}
 	resp.Header.Del(headerKey)
 
+	numPreloads := 0
 	for _, value := range values {
 		links, err := httplink.Parse(value)
 		if err != nil {
 			log.Printf("warning: %v -- this header was ignored", err)
 			continue
 		}
 		for _, link := range links {
-			if link.IsPreload() {
+			if numPreloads < maxNumPreloads && link.IsPreload() {
```

Comment:
I think you want this condition to be nested; the else fallback is not desired. (In the test, 21 should not get a link header at all.)

Reply:
Done.